src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/StegService.kt

```diff
@@ -119,6 +119,25 @@ class StegService(
         }
     }
 
+    @Transactional
+    fun tilbakeførBehandlingSteg(behandling: Behandling, tilbakeføresSteg: BehandlingSteg) {
+        val nåværendeBehandlingSteg = behandling.steg
+        if (nåværendeBehandlingSteg.sekvens < tilbakeføresSteg.sekvens) {
+            throw Feil(
+                "Behandling ${behandling.id} er på ${nåværendeBehandlingSteg.visningsnavn()}, " +
+                    "kan ikke tilbakeføres til ${tilbakeføresSteg.visningsnavn()}."
+            )
+        }
+        behandling.behandlingStegTilstand.forEach {
+            when (it.behandlingSteg) {
+                nåværendeBehandlingSteg -> it.behandlingStegStatus = BehandlingStegStatus.TILBAKEFØRT
+                tilbakeføresSteg -> it.behandlingStegStatus = BehandlingStegStatus.KLAR
+                else -> {} // gjør ingenting
+            }
+        }
+        behandlingRepository.saveAndFlush(oppdaterBehandlingStatus(behandling, tilbakeføresSteg))
+    }
+
```

Comment:
Skal man kunne gå flere enn ett steg tilbake? Kan man f.eks gå fra VEDTAK tilbake til REGISTRERE_SØKNAD? Blir det da riktig at kun status til VEDTAK og status til REGISTRERE_SØKNAD skal endres? Hva med stegene mellom? Her blir vel alle stegene mellom stående som UTFØRT?

Reply:
@bragejahren Ja, SB kan gå fra VEDTAK til REGISTERE_SØKNAD steg, men da kaller vi ikke denne metoden. Når det skjer, oppdateres alle stegene mellom VEDTAK og REGISTERE_SØKNAD til tilbakeført og REIGSTERE_SØKNAD steg til KLAR.  Men denne metoden kommer vi til å bruke for å flytte fra nåværende steg til et annet steg uten å utføre dette steget. F.eks Saken er på BESLUTTE_VEDTAK steg og SB har underkjente saken. Da bør saken flyttes tilbake til VEDTAK steg. Da blir VEDTAK steg KLAR og BESLUTTE_VEDTAK TILBAKEFØRT.